### PR TITLE
DYN-4241-Step6-PackageDependencies

### PR DESCRIPTION
**Pull request by RobertGlobant20**
### Purpose

I added the Step 6 to the packages guide, the automation needed for opening the PackageDetaiils and adding support for showing the PackageDetails User Control over the dark overlay.
Basically I did the next changes:

- The highlight and cutoff functionality was moved from the Guide class to the Step class, so it will be executed on Step.Show(), Step.Hide() or when exiting the Guide.
- I modified resources for the Package Details popup and remove the resource for the "Autodesk Sample Package" due that it shouldn't be localized.
- I added some code to subscribe the "Search for a Package" menuitem Click event to the NextStep method (bug reported by Sol).
- I added a method that will be executing the search in the PackageSearch window after the initial search is completed so the message "Please wait.." is shown.
- I disable the Back button in the Step1 due that was closing all the Steps and leaving the dark overlay.
- Finally I added some methods for closing the PackageSearch window

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Adding the Step 6 (package dependencies) to the Packages Guide

### Reviewers

@QilongTang 

### FYIs

@filipeotero

### CODE REVIEW

**Comment by RobertGlobant20** [comment]
This is how the Packages guide looks until Step6 (Package Dependencies)
packages_guide_step6
.

File: src/DynamoCoreWpf/UI/GuidedTour/GuidesValidationMethods.cs [hunk 1]
```diff
@@ -152,41 +157,122 @@ internal static void ExecutePackageSearch(Step stepInfo, StepUIAutomation uiAuto
...
-                }             
+
+                PackageManagerSearchView packageManager = Guide.FindWindowOwned(stepInfo.HostPopupInfo.WindowName, stepInfo.MainWindow as Window) as PackageManagerSearchView;
+                if (packageManager == null)
+                    return;
+                PackageManagerSearchViewModel packageManagerViewModel = packageManager.DataContext as PackageManagerSearchViewModel;
+                if (packageManagerViewModel == null)
+                    return;
+
+                //Due that we need to search the Autodesk Sample package after the inicial search is completed
```
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/GuidedTour/GuidesValidationMethods.cs:170]: typo `initial`

File: src/DynamoCoreWpf/UI/GuidedTour/Step.cs [hunk 2]
```diff
@@ -246,6 +252,92 @@ internal void UpdatePlacementTarget()
...
+            else
+            {
+                StepGuideBackground.ClearCutOffSection();
+            }
+        }
+
+        /// <summary>
+        /// This method will set the highlight rectangle color if there is any configured in the json file
+        /// </summary>
+        /// <param name="bVisible">It will say it the Highlight Area will be disabled or enable</param>
```
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/GuidedTour/Step.cs:299]: `if the Highlight Area will be disabled or enable`? Please update

File: src/DynamoCoreWpf/UI/GuidedTour/Step.cs [hunk 3]
```diff
@@ -246,6 +252,92 @@ internal void UpdatePlacementTarget()
...
+                return;
+            HostPopupInfo.HostUIElement = hostUIElement;
+            stepUIPopup.PlacementTarget = hostUIElement;
+            UpdateLocation();
+        }
+
+        /// <summary>
+        /// This method will update the CutOff rectangle size everytime that the step change
+        /// </summary>
+        /// <param name="bVisible">It will say it the CutOff Area will be disabled or enable</param>
```
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/GuidedTour/Step.cs:267]: same comment here

File: src/DynamoCoreWpf/UI/GuidedTour/GuidesValidationMethods.cs (hunk 1 above)
Comment (RobertGlobant20) [reply src/DynamoCoreWpf/UI/GuidedTour/GuidesValidationMethods.cs]: @QilongTang fixed in the next commit:
https://github.com/DynamoDS/Dynamo/pull/12277/commits/7b9f128012709949425ccbf45f7bf3f52d19c89a

File: src/DynamoCoreWpf/UI/GuidedTour/Step.cs (hunk 2 above)
Comment (RobertGlobant20) [reply src/DynamoCoreWpf/UI/GuidedTour/Step.cs]: @QilongTang fixed in the next commit:
https://github.com/DynamoDS/Dynamo/pull/12277/commits/7b9f128012709949425ccbf45f7bf3f52d19c89a

File: src/DynamoCoreWpf/UI/GuidedTour/Step.cs (hunk 3 above)
Comment (RobertGlobant20) [reply src/DynamoCoreWpf/UI/GuidedTour/Step.cs]: @QilongTang fixed in the next commit:
https://github.com/DynamoDS/Dynamo/pull/12277/commits/7b9f128012709949425ccbf45f7bf3f52d19c89a

**Review by QilongTang — APPROVED** [review]

**Pull request merged**